invenio/modules/search/views/search.py

```diff
@@ -248,12 +248,21 @@ def search(collection, p, of, ot, so, sf, sp, rm, rg, jrec):
     })
 
     # FIXME refactor to separate search hook
+    filtered_facets = ''
     from invenio.modules.search.walkers.elasticsearch import ElasticSearchDSL
     if 'post_filter' in request.values:
         post_filter = Query(request.values.get('post_filter')).query.accept(
             ElasticSearchDSL()
         )
         response.body['post_filter'] = post_filter
+        # extracting the facet filtering
+        from invenio.modules.search.walkers.facets import FacetsVisitor
+        filtered_facets = Query(request.values.get('post_filter')).query.accept(
```

Comment:
you can reuse parsed query from https://github.com/inveniosoftware/invenio/pull/3400/files#diff-66b51f83bbfcce07b1d663fe0cf19b92R254